app/services/plans/create_service.rb

```diff
@@ -52,13 +52,8 @@ def call
         end
 
         if args[:charges].present?
-          args[:charges].each do |charge|
-            new_charge = create_charge(plan, charge)
-
-            if charge[:tax_codes].present?
-              taxes_result = Charges::ApplyTaxesService.call(charge: new_charge, tax_codes: charge[:tax_codes])
-              taxes_result.raise_if_error!
-            end
+          args[:charges].each do |charge_params|
+            Charges::CreateService.call!(plan:, params: charge_params)
```

Comment:
good catch removing duplicated code already on a service that should be the way to create the charges.